src/modules/27-platform/secrets/components/CreateUpdateSecret/__tests__/CreateUpdateSecret.test.tsx

```diff
@@ -70,9 +70,37 @@ describe('CreateUpdateSecret', () => {
 
     await waitFor(() => expect(getByText('platform.secrets.labelValue')).toBeTruthy())
 
+    const secretManagerField = screen.getByTestId(/secretManagerIdentifier/)
+    expect(secretManagerField).toBeInTheDocument()
+    expect(secretManagerField).not.toBeDisabled()
+
     expect(container).toMatchSnapshot()
   })
 
+  test('it should disable secret manager field when connectorTypeContext is of secret manager type and secret type is of text or file type', async () => {
+    const { getByText } = render(
+      <TestWrapper path="/account/:accountId/resources/secrets" pathParams={{ accountId: 'dummy' }}>
+        <CreateUpdateSecret type={'SecretText'} connectorTypeContext="GcpSecretManager" />
+      </TestWrapper>
+    )
+    await waitFor(() => expect(getByText('platform.secrets.labelValue')).toBeInTheDocument())
+    expect(getByText('platform.secrets.labelSecretName')).toBeInTheDocument()
+    const secretManagerField = screen.getByTestId(/secretManagerIdentifier/)
+    expect(secretManagerField).toBeInTheDocument()
+    expect(secretManagerField).toBeDisabled()
```

Comment:
These tests are a bit difficult to read. A good method to make them a bit more visually scannable is to use some white space to break up the setup/action/assert parts.
```suggestion
    render(
      <TestWrapper path="/account/:accountId/resources/secrets" pathParams={{ accountId: 'dummy' }}>
        <CreateUpdateSecret type="SecretText" connectorTypeContext="GcpSecretManager" />
      </TestWrapper>
    )
    
    expect(await screen.findByText('platform.secrets.labelValue')).toBeInTheDocument()
    expect(screen.getByText('platform.secrets.labelSecretName')).toBeInTheDocument()
    expect(screen.getByTestId(/secretManagerIdentifier/)).toBeDisabled()
```

Also, try to stick to using `screen.` instead of the returned functions and try `findBy*` instead of `waitFor`.